File: Core/ChatService/Resources/MessegeReadResource.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Data.Domain;

namespace Core.ChatService.Resources
{
    public class MessageReadResource
    {
        public Guid Id { get; set; }
        public DateTime CreationDateTime { get; set; }
        public DateTime? DeleteTime { get; set; }
        public Guid? DeleterId { get; set; }
        
        public Guid SenderId { get; set; }
        
        public Guid ReciverId { get; set; }
        
        public String Context { get; set; }
        
    }
}

File: Web/ClientApp/src/app/web-app/MessegeReadResource.ts
import { User } from "./User";
import { Chat } from "./Chat";
export interface MessegeReadResource {
    id: string;
    creationDateTime: Date | string;
    deleteTime: Date | string | null;
    deleterId: string | null;
    deleter: User;
    senderId: string;
    sender: User;
    reciverId: string;
    reciver: User;
    context: string;
    chatId: string;
    chat: Chat;
}

